COMPONENT-GUIDELINES.md

```diff
@@ -0,0 +1,293 @@
+
+### Interface & Service Props
+
+In the interest of consistency, use the following naming patterns when defining interface and service props in source components:
+
+| **Scenario**      | **Recommended Prop Variable Name** |
+| ----------------- | ---------------------------------- |
+| $.interface.http  | `http`                             |
+| $.interface.timer | `timer`                            |
+| $.service.db      | `db`                               |
+
+## Sources
+
+### Webhook vs Polling Sources
+
+Create subscription webhooks sources (vs polling sources) whenever possible. Webhook sources receive/emit events in real-time and typically use less compute time from the user’s account. Note: In some cases, it may be appropriate to support webhook and polling sources for the same event. For example, Calendly supports subscription webhooks for their premium users, but non-premium users are limited to the REST API. A webhook source can be created to emit new Calendly events for premium users, and a polling source can be created to support similar functionality for non-premium users.
+
+### Source Name
+
+Source name should be a singular, title-cased name and should start with "New" (unless emits are not limited to new items). Name should not be slugified and should not include the app name. 
+NOTE: Pipedream does not currently distinguish real-time event sources for end-users automatically. The current pattern to identify a real-time event source is to include “(Instant)” in the source name. This will change in the future. E.g., “New Search Mention” or “New Submission (Instant)”.
+
+### Source Description 
+
+Enter a short-description that provides more detail than the name alone. Typically starts with "Emit new". E.g., “Emit new Tweets that matches your search criteria”.
+
+### Emit a Summary
+
+Always [emit a summary](COMPONENT-API.md#emit) for each event. For example, the summary for each new Tweet emitted by the Search Mentions source is the content of the Tweet itself.
+
+If no sensible summary can be identified, submit the event payload in string format as the summary.
+
+### Deduping
+
+Use built-in deduping strategies whenever possible (unique, greatest, last) vs developing custom deduping code. Develop custom deduping code if the existing strategies do not support the requirements for a source.
+
+### Polling Sources 
+
+#### Emit Events on First Run
+
+Polling sources should emit events on the first run. This helps users to know their source works when they activate it. This also provides users with events they can immediately use to support workflow development.
+
+#### Rate Limit Optimization
+
+When building a polling source, cache the most recently processed ID using $.service.db whenever the API accepts a “since_id” (or equivalent). Some apps (e.g., Github) do not count requests that do not return new results against a user’s API quota.
+
+### Webhook Sources
+
+#### Helper Methods
```

Comment:
I think it's worth first describing `hooks` in this section and linking to those API docs, noting that that's how subscription webhooks are created / destroyed.

Reply:
done